### PR TITLE
OT-318 OT-321 Update Chat and Message UI

### PR DESCRIPTION
**Pull request by Boehrsi**
**Describe what the problem was / what the new feature is**

- Reworked chat message UI
- Generic file attachments now can have text
- Status messages for encryption state changes
- Bugfixes

### CODE REVIEW

File: lib/src/message/message_builder.dart [hunk 1]
```diff
@@ -263,31 +331,57 @@ class MessagePartState extends StatelessWidget {
...
+        ),
+      ),
+    );
+  }
+}
+
+EdgeInsetsGeometry getNamePaddingForGroups(BuildContext context) {
+  if (MessageData.of(context).isGroup) {
+    return EdgeInsets.only(
+      top: 2.0,
```
Comment (florianhaar) [inline lib/src/message/message_builder.dart]: hardcoded dimension

File: lib/src/message/message_builder.dart [hunk 2]
```diff
@@ -263,31 +331,57 @@ class MessagePartState extends StatelessWidget {
...
-bool isImage(BuildContext context) {
-  final attachment = MessageData.of(context).attachment;
-  return attachment != null && attachment.type == ChatMsg.typeImage;
+class MessagePartFlag extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Visibility(
+      visible: MessageData.of(context).isFlagged,
+      child: Padding(
+        padding: EdgeInsets.only(top: 8.0, right: 4.0, left: 4.0),
```
Comment (florianhaar) [inline lib/src/message/message_builder.dart]: hardcoded dimension

File: lib/src/message/message_builder.dart [hunk 3]
```diff
@@ -263,31 +331,57 @@ class MessagePartState extends StatelessWidget {
...
-            size: 14.0,
+            size: 16.0,
             color: MessageData.of(context).secondaryTextColor,
           ),
         );
         break;
       default:
-        return Container();
+        return Container(
+          width: 20.0,
```
Comment (florianhaar) [inline lib/src/message/message_builder.dart]: hardcoded dimension

File: lib/src/message/message_builder.dart [hunk 4]
```diff
@@ -263,31 +331,57 @@ class MessagePartState extends StatelessWidget {
...
         );
         break;
       case ChatMsg.messageStateReceived:
         return Padding(
-          padding: EdgeInsets.only(left: iconTextPadding),
+          padding: EdgeInsets.only(top: 10.0, left: iconTextPadding),
           child: Icon(
             Icons.done_all,
-            size: 14.0,
+            size: 16.0,
```
Comment (florianhaar) [inline lib/src/message/message_builder.dart]: hardcoded dimension

File: lib/src/message/message_builder.dart [hunk 5]
```diff
@@ -263,31 +331,57 @@ class MessagePartState extends StatelessWidget {
...
             color: MessageData.of(context).secondaryTextColor,
           ),
         );
         break;
       case ChatMsg.messageStateReceived:
         return Padding(
-          padding: EdgeInsets.only(left: iconTextPadding),
+          padding: EdgeInsets.only(top: 10.0, left: iconTextPadding),
```
Comment (florianhaar) [inline lib/src/message/message_builder.dart]: hardcoded dimension

File: lib/src/message/message_builder.dart [hunk 6]
```diff
@@ -263,31 +331,57 @@ class MessagePartState extends StatelessWidget {
     switch (MessageData.of(context).state) {
       case ChatMsg.messageStateDelivered:
         return Padding(
-          padding: EdgeInsets.only(left: iconTextPadding),
+          padding: EdgeInsets.only(top: 10.0, left: iconTextPadding),
           child: Icon(
             Icons.done,
-            size: 14.0,
+            size: 16.0,
```
Comment (florianhaar) [inline lib/src/message/message_builder.dart]: hardcoded dimension

File: lib/src/chat/chat.dart [hunk 7]
```diff
@@ -421,18 +421,23 @@ class _ChatState extends State<Chat> with ChatComposer, ChatCreateMixin, InviteM
   ListView buildListItems(MessagesStateSuccess state) {
     return ListView.custom(
       controller: _scrollController,
-      padding: new EdgeInsets.all(listItemPadding),
+      padding: new EdgeInsets.all(8.0),
```
Comment (florianhaar) [inline lib/src/chat/chat.dart]: hardcoded dimension

File: lib/src/message/message_received.dart [hunk 8]
```diff
@@ -82,52 +71,59 @@ class MessageReceived extends StatelessWidget {
...
+          children: <Widget>[
+            Visibility(
+              visible: isGroupChat,
+              child: Padding(
+                padding: const EdgeInsets.only(right: messagesHorizontalPadding),
+                child: Avatar(
+                  textPrimary: name,
+                  textSecondary: email,
+                  color: color,
+                  size: 34.0,
```
Comment (florianhaar) [inline lib/src/message/message_received.dart]: hardcoded dimension

File: lib/src/message/message_received.dart [hunk 9]
```diff
@@ -82,52 +71,59 @@ class MessageReceived extends StatelessWidget {
...
                 child: Column(
                   mainAxisSize: MainAxisSize.min,
                   crossAxisAlignment: CrossAxisAlignment.start,
                   children: <Widget>[
-                    isGroupChat ? Text(name, style: TextStyle(color: color)) : Container(constraints: BoxConstraints(maxWidth: zero)),
+                    isGroupChat
+                        ? Padding(
+                            padding: EdgeInsets.only(
+                                top: messagesVerticalPadding,
+                                bottom: 2.0,
```
Comment (florianhaar) [inline lib/src/message/message_received.dart]: hardcoded dimension

**Comment by Boehrsi** [comment]
The requested changes will be implemented in a separate ticket, as we plan to clean up our dimensions file anyway.

**Pull request merged**